src/main/routes/instance/typingRoutes.ts

```diff
@@ -18,44 +18,49 @@
  */
 
 import * as express from 'express';
-import {check, validationResult} from 'express-validator/check';
+import * as Joi from 'joi';
 import InstanceService from '../../InstanceService';
 
 export interface TypingMessageRequest {
   conversationId: string;
   payload: 'started' | 'stopped';
 }
 
+const joiSchema = {
+  conversationId: Joi.string()
+    .uuid()
+    .required(),
+  payload: Joi.string()
+    .regex(/^(started|stopped)$/)
+    .required(),
+};
+
 const typingRoutes = (instanceService: InstanceService): express.Router => {
   const router = express.Router();
 
-  router.post(
-    '/api/v1/instance/:instanceId/typing',
-    [check('conversationId').isUUID(), check('payload').matches(/^(started|stopped)$/)],
-    async (req: express.Request, res: express.Response) => {
-      const {instanceId = ''}: {instanceId: string} = req.params;
-      const {conversationId, payload}: TypingMessageRequest = req.body;
-
-      const errors = validationResult(req);
-      if (!errors.isEmpty()) {
-        return res.status(422).json({errors: errors.mapped()});
-      }
-
-      if (!instanceService.instanceExists(instanceId)) {
-        return res.status(400).json({error: `Instance "${instanceId}" not found.`});
-      }
-
-      try {
-        const instanceName = await instanceService.sendTyping(instanceId, conversationId, payload);
-        return res.json({
-          instanceId,
-          name: instanceName,
-        });
-      } catch (error) {
-        return res.status(500).json({error: error.message, stack: error.stack});
-      }
+  router.post('/api/v1/instance/:instanceId/typing', async (req: express.Request, res: express.Response) => {
+    const {instanceId = ''}: {instanceId: string} = req.params;
+    const {conversationId, payload}: TypingMessageRequest = req.body;
+
+    const {error: joiError} = Joi.validate(req.body, joiSchema);
+    if (joiError) {
+      return res.status(422).json({error: `Validation error: ${joiError.message}}`});
```

Comment:
I see this piece of code duplicated 5 or 6 times. The only thing that changes is the schema that is given to the validation.

Probably we can refactor that into a middleware/higher order function/whatever